### DISCUSSION TITLE
Is it possible to write custom values to lane_context/SharedValues?

### DISCUSSION THREAD

**Discussion by blwinters** [fastlane/fastlane#21299] [Q&A]
I'm trying to pass the updated build number between two lanes in order to log it upon upload success. I've scoured every relevant discussion I could find, but so for I haven't been able to identify the syntax for adding a custom value to the lane_context hash. The [documentation for lane_context](https://docs.fastlane.tools/advanced/lanes/#lane-context) doesn't explicitly say that the hash is read only, so is that the case?

[This discussion](https://github.com/fastlane/fastlane/issues/14255#issuecomment-471986433) suggests that ENV values are more for inputs where lane_context is for communication between actions, which makes sense to me, but I haven't been able to figure out how to write values to the lane_context. Can anyone clarify the intended behavior? 

I can easily use ENV values instead of lane_context, but it would be nice to have an explanation and updated documentation. I'd be happy to create a PR for documentation once I know what it should be.

Here is a simple example I have in my fastfile to test the two approaches:
``` ruby
# errors
  desc "Test logging a lane context value"
  lane :test_lane_context do
    latest_build_number = 17
    Actions.lane_context[SharedValues::NEW_BUILD_NUMBER] = "#{(latest_build_number + 1)}"
    log_lane_context
  end

  private_lane :log_lane_context do
    UI.success "Did upload build #{lane_context[SharedValues::NEW_BUILD_NUMBER]} to TestFlight"
  end

# succeeds
  desc "Test logging an ENV value"
  lane :test_env do
    latest_build_number = 17
    ENV["NEW_BUILD_NUMBER"] = "#{(latest_build_number + 1)}"
    log_env
  end

  private_lane :log_env do
    UI.success "Did upload build #{ENV["NEW_BUILD_NUMBER"]} to TestFlight"
  end

```

**Comment by enoktate** [comment]
**Update:** It turns out that it wasn't working in the CI environment due to another issue in my code that was unrelated to Fastlane. After resolving it, the implementation below now works for me.

---

I'm currently dealing with the same issue. Initially, I thought I had resolved it because it functioned as expected on my local machine. ~However, when the same lane is triggered with the same parameters in the GitHub Actions CI environment, it fails to work.~

Here's how I write to `lane_context` in one lane:
``` Ruby
    lane_context[:JIRA_IDS_ARRAY] = jira_ids_array
```

And this is how I read from `lane_context` in another lane:
``` Ruby
    jira_ids_array = lane_context[:JIRA_IDS_ARRAY]
```

I'm unsure if this approach is fundamentally incorrect. ~I can't explain why it works locally but not in the CI environment.~
